services/cd-service/pkg/repository/repository.go

```diff
@@ -534,22 +534,26 @@ func (r *repository) useRemote(ctx context.Context, callback func(*git.Remote) e
 }
 
 func (r *repository) drainQueue() []transformerBatch {
+	if r.config.MaximumCommitsPerPush == 0 {
```

Comment:
nitpick: from the naming it's a bit counter intuitive that 0 is valid value.
Maybe `maximumAdditionalCommitsPerPush`?

Reply:
Yes, agreed. I changed the code so that the default is set when we initialize the repository.